rtsp_to_hls/lib/pipeline.ex

```diff
@@ -13,72 +13,65 @@ defmodule Membrane.Demo.RtspToHls.Pipeline do
     spec = [
       child(:source, %Membrane.RTSP.Source{
         transport: {:udp, options.port, options.port + 5},
-        allowed_media_types: [:video],
-        stream_uri: options.stream_url
-      }),
-      child(
-        :hls,
-        %Membrane.HTTPAdaptiveStream.SinkBin{
-          target_window_duration: Membrane.Time.seconds(120),
-          manifest_module: Membrane.HTTPAdaptiveStream.HLS,
-          storage: %Membrane.HTTPAdaptiveStream.Storages.FileStorage{
-            directory: options.output_path
-          }
-        }
-      )
+        allowed_media_types: [:video, :audio],
+        stream_uri: options.stream_url,
+        on_connection_closed: :send_eos
+      })
     ]
 
     {[spec: spec],
      %{
-       video: nil,
        output_path: options.output_path,
        parent_pid: options.parent_pid,
-       rtp_started: false
+       tracks_left_to_link: nil,
+       track_specs: []
      }}
   end
 
   @impl true
-  def handle_child_notification(
-        {:new_track, ssrc, %{type: :video} = track},
-        :source,
-        _ctx,
-        %{rtp_started: false} = state
-      ) do
-    Logger.debug(":new_rtp_stream")
+  def handle_child_notification({:set_up_tracks, tracks}, :source, _ctx, state) do
+    tracks_left_to_link =
+      [:audio, :video]
+      |> Enum.filter(fn media_type -> Enum.any?(tracks, &(&1.type == media_type)) end)
```

Comment:
So we just support one audio and one video track? If so, I'd just add `audio_track` and `video_track` fields to the state, it should be much simpler

Reply:
I don't really think so, since we support also only video (and only audio) streams, I tried doing this and a lot of branching arises